lib/staccato/hit.ex

```diff
@@ -42,10 +42,10 @@ defmodule Staccato.Hit do
   def transaction(tracker, options), do: Staccato.Hit.Transaction.build(tracker, options)
   def transaction_item(tracker, options), do: Staccato.Hit.TransactionItem.build(tracker, options)
 
-  def track!(hit) do
+  def track!(hit, user_agent) do
```

Comment:
Rather than passing user_agent into `track!` it should probably be in the configuration options on the tracker itself. Then post has direct access to it.

Reply:
Good feedback, can update that.

Reply:
Is there an ideal portion of the config maps where you're passing other things that are assigned as headers to the request which this can move cleanly to?

Reply:
So, going back into my project (ad impression tracking software), I realized what I'm doing is I have a singleton tracker in a GenServer that I cast requests to as impressions are created in our database. For each impression, the User Agent varies as the impressions are created by a different consumer of the ad every time. For this reason, I'm not sure that the user_agent in the config makes sense for how we've implemented this (or possibly how its likely to be used by others. Would you have any other suggestions for how to handle this? 

Offending piece of code here: https://github.com/gitcoinco/codefund/blob/70bb2d2d43c0c1dd3e125391d9c13e595f0638a2/lib/ad_service/tracking/analytics_manager.ex

Reply:
Oh, in that case you probably want to set the `user_agent` on the hit: https://github.com/tpitale/staccato-ex/blob/master/lib/staccato/hit/global.ex#L11

Sorry that isn't documented yet.

Reply:
No worries whatsoever, thanks!